amy-cohen/public/scripts/article.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+var app = app || {};
+
+//is this how? Nothing else is working and there is so much online that shows different options. Hoping when you said wrap it that you meant the whole thing.
+(function (module) {
+
+function Article(rawDataObj) {
+  // REVIEW: In Lab 8, we explored a lot of new functionality going on here. Let's re-examine the concept of context. Normally, "this" inside of a constructor function refers to the newly instantiated object. However, in the function we're passing to forEach, "this" would normally refer to "undefined" in strict mode. As a result, we had to pass a second argument to forEach to make sure our "this" was still referring to our instantiated object. One of the primary purposes of lexical arrow functions, besides cleaning up syntax to use fewer lines of code, is to also preserve context. That means that when you declare a function using lexical arrows, "this" inside the function will still be the same "this" as it was outside the function. As a result, we no longer have to pass in the optional "this" argument to forEach!
+  Object.keys(rawDataObj).forEach(key => this[key] = rawDataObj[key]);
+};
+
+Article.all = [];
+
+Article.prototype.toHtml = function() {
+  var template = Handlebars.compile($('#article-template').text());
+
+  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+  this.publishStatus = this.publishedOn ? `published ${this.daysAgo} days ago` : '(draft)';
+  this.body = marked(this.body);
+
+  return template(this);
+};
+
+Article.loadAll = articleData => {
+  articleData.sort((a,b) => (new Date(b.publishedOn)) - (new Date(a.publishedOn)))
+
+  /* OLD forEach():
+  articleData.forEach(articleObject => Article.all.push(new Article(articleObject)));
```

Comment:
The benefit of this old forEach is that it pushed things into `Article.all`; the reason your code doesn't put articles on the page right now is that you call `map` on the next line, but don't assign that to `Article.all`. You probably mean to say:

`Article.all = articleData.map....`